src/main/resources/openapi.json

```diff
@@ -1,286 +1,935 @@
+                ],
+                "description": "Assigns one or more topic partitions to a consumer.",
+                "operationId": "assign",
+                "requestBody": {
+                    "description": "List of topic partitions to assign to the consumer.",
+                    "content": {
+                        "application/json": {
+                            "schema": {
+                                "$ref": "#/components/schemas/PartitionsForPositions"
+                            }
+                        }
+                    },
+                    "required": true
+                },
+                "responses": {
+                    "204": {
+                        "description": "OK"
+                    },
+                    "404": {
+                        "description": "The specified consumer instance was not found."
+                    },
+                    "409": {
+                        "description": "Subscription to topics, partitions, and pattern are mutually exclusive."
+                    }
+                }
+            },
+            "parameters": [
+                {
+                    "name": "groupid",
+                    "in": "path",
+                    "description": "ID of the consumer group to which the consumer belongs.",
+                    "required": true,
+                    "schema": {
+                        "type": "string"
+                    }
+                },
+                {
+                    "name": "name",
+                    "in": "path",
+                    "description": "Name of the consumer to which you want to assign topic partitions.",
+                    "required": true,
+                    "schema": {
+                        "type": "string"
+                    }
+                }
+            ]
         }
-      ]
     },
-    "/consumers/{groupid}/instances/{name}/partitions/end": {
-      "post": {
-        "operationId": "seekToEnd",
-        "responses": {
-          "200": {
-            "description": "OK"
-          }
+    "components": {
+        "schemas": {
+            "ApiResponse": {
+                "type": "object",
+                "properties": {
+                    "code": {
+                        "format": "int32",
+                        "type": "integer"
+                    },
+                    "type": {
+                        "type": "string"
+                    },
+                    "message": {
+                        "type": "string"
+                    }
+                }
+            },
+            "Consumer": {
+                "title": "Root Type for Consumer",
+                "description": "The root of the Consumer type's schema.",
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "format": {
+                        "type": "string"
```

Comment:
Can you document the valid values for this and other properties?

Reply:
@laidan6000 @PaulRMellor can you create some descriptions there, please?

Reply:
@stanlyDoge - picking up and will add directly in apicurio